torch/__init__.py

```diff
@@ -1775,20 +1775,22 @@ def registerOp(cls, op_key, full_schema, op_impl, dispatch_key):
     "onnx",
 }
 
-def __getattr__(name):
-    # Deprecated attrs
-    replacement = _deprecated_attrs.get(name)
-    if replacement is not None:
-        import warnings
-        warnings.warn(f"'{name}' is deprecated, please use '{replacement.__module__}.{replacement.__name__}()'", stacklevel=2)
-        return replacement()
-
-    # Lazy modules
-    if name in _lazy_modules:
-        import importlib
-        return importlib.import_module(f".{name}", __name__)
-
-    raise AttributeError(f"module '{__name__}' has no attribute '{name}'")
+
+if not TYPE_CHECKING:
+    def __getattr__(name):
+        # Deprecated attrs
+        replacement = _deprecated_attrs.get(name)
+        if replacement is not None:
+            import warnings
+            warnings.warn(f"'{name}' is deprecated, please use '{replacement.__module__}.{replacement.__name__}()'", stacklevel=2)
+            return replacement()
+
+        # Lazy modules
+        if name in _lazy_modules:
+            import importlib
+            return importlib.import_module(f".{name}", __name__)
+
+        raise AttributeError(f"module '{__name__}' has no attribute '{name}'")
```

Comment:
It would be nice to merge it with `if TYPE_CHECKING:` condition in `__init__.py` so that it would like like
```python
if TYPE_CHECKING:
  import torch.onnx
  import torch.compile
else:
 def __getattr__:
   ...

Reply:
Thanks, updated!